internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go

```diff
@@ -2,14 +2,47 @@ package fsm
 
 import (
 	"context"
+	"encoding/json"
 
 	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func sFnDeleteShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	if !isGardenerCloudDelConfirmationSet(s.shoot.Annotations) {
```

Comment:
Add following at the beginning of the function for better logging

	m.log.Info("Delete shoot state")